packages/webview_flutter_lwe/lib/src/lwe_webview_controller.dart

```diff
@@ -0,0 +1,405 @@
+      case proxyAuthentication:
+        return WebResourceErrorType.proxyAuthentication;
+      case connect:
+        return WebResourceErrorType.connect;
+      case io:
+        return WebResourceErrorType.io;
+      case timeout:
+        return WebResourceErrorType.timeout;
+      case redirectLoop:
+        return WebResourceErrorType.redirectLoop;
+      case unsupportedScheme:
+        return WebResourceErrorType.unsupportedScheme;
+      case failedSSLHandshake:
+        return WebResourceErrorType.failedSslHandshake;
+      case badURL:
+        return WebResourceErrorType.badUrl;
+      case file:
+        return WebResourceErrorType.file;
+      case fileNotFound:
+        return WebResourceErrorType.fileNotFound;
+      case tooManyRequests:
+        return WebResourceErrorType.tooManyRequests;
+    }
+
+    throw ArgumentError(
+      'Could not find a WebResourceErrorType for errorCode: $errorCode',
+    );
+  }
+}
+
+/// A place to register callback methods responsible to handle navigation events
+/// triggered by the [LweWebView].
+class LweNavigationDelegate extends PlatformNavigationDelegate {
+  /// Creates a new [LweNavigationDelegate].
+  LweNavigationDelegate(super.params) : super.implementation();
+
+  late final MethodChannel _navigationDelegateChannel;
+  PageEventCallback? _onPageFinished;
+  PageEventCallback? _onPageStarted;
+  ProgressCallback? _onProgress;
+  WebResourceErrorCallback? _onWebResourceError;
+  NavigationRequestCallback? _onNavigationRequest;
+
+  /// Called when [TizenView] is created.
+  void onCreate(int viewId) {
+    _navigationDelegateChannel =
+        MethodChannel(kLweNavigationDelegateChannelName + viewId.toString());
+    _navigationDelegateChannel.setMethodCallHandler((MethodCall call) async {
+      final Map<String, Object?> arguments =
+          (call.arguments as Map<Object?, Object?>).cast<String, Object?>();
+      switch (call.method) {
+        case 'navigationRequest':
+          return _handleNavigation(arguments['url']! as String,
+              isForMainFrame: arguments['isForMainFrame']! as bool);
+        case 'onPageFinished':
+          if (_onPageFinished != null) {
+            _onPageFinished!(arguments['url']! as String);
+          }
+          return null;
+        case 'onProgress':
+          if (_onProgress != null) {
+            _onProgress!(arguments['progress']! as int);
+          }
+          return null;
+        case 'onPageStarted':
+          if (_onPageStarted != null) {
+            _onPageStarted!(arguments['url']! as String);
+          }
+          return null;
+        case 'onWebResourceError':
+          if (_onWebResourceError != null) {
+            _onWebResourceError!(LweWebResourceError._(
+              errorCode: arguments['errorCode']! as int,
+              description: arguments['description']! as String,
+              failingUrl: arguments['failingUrl']! as String,
+              isForMainFrame: true,
```

Comment:
Now that the error code to type conversion is done by the Dart class, the `ErrorCodeToString` method and the related logic can be removed from the C++ implementation.

Reply:
you are right. I removed the code. and I will remove same code on webview_flutter_tizen.